mettle/worker/worker.go

```diff
@@ -462,20 +462,29 @@ func (w *worker) runTask(msg *pubsub.Message) *pb.ExecuteResponse {
 		WorkerStartTimestamp: ptypes.TimestampNow(),
 	}
 	w.taskStartTime = time.Now()
-	req, action, command, status := w.readRequest(msg.Body)
+	req, err := readRequest(msg.Body)
+	if err != nil {
+		log.Error("Bad request: %s", err)
+		return &pb.ExecuteResponse{
+			Result: &pb.ActionResult{},
+			Status: status(codes.FailedPrecondition, err.Error()),
+		}
+	}
 	if req != nil {
 		w.actionDigest = req.ActionDigest
 	}
```

Comment:
I think you can get rid of that if block now. Since if `err` is nil, `req` should not be.

Reply:
good catch thanks